### PR TITLE
Adds missing parameter with required value directive for Java API (#2025)

### PR DESCRIPTION
**Pull request by nikom1337**
and updates documentation

Refs: #2025

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

File: akka-http/src/main/scala/akka/http/javadsl/server/Rejections.scala [hunk 1]
```diff
@@ -57,6 +57,16 @@ trait MissingQueryParamRejection extends Rejection {
   def parameterName: String
 }
 
+/**
+ * Rejection created by parameter filters.
+ * Signals that the request was rejected because a query parameter value was not equal to required one.
+ */
+trait InvalidRequiredValueForQueryParamRejection extends Rejection {
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/javadsl/server/Rejections.scala]: Perhaps mark with `@DoNotInherit`?

File: akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala [hunk 2]
```diff
@@ -200,6 +200,10 @@ object RejectionHandler {
         case MissingQueryParamRejection(paramName) ⇒
           complete((NotFound, "Request is missing required query parameter '" + paramName + '\''))
       }
+      .handle {
+        case InvalidRequiredValueForQueryParamRejection(paramName, requiredValue, actualValue) ⇒
+          complete((NotFound, s"Request is missing required value '$requiredValue' for query parameter '$paramName', actualValue '$actualValue'"))
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala]: I think echo'ing a user-provided value in a response is considered a potential security risk (harmless on its own, but could be used in elaborate attach contexts), so perhaps best not to include the `actualValue`?

File: akka-http/src/main/scala/akka/http/javadsl/server/Rejections.scala (hunk 1 above)
Comment (nikom1337) [reply akka-http/src/main/scala/akka/http/javadsl/server/Rejections.scala]: I can mark it as a `@DoNotInherit`, but what with the rest of a traits in Rejections file? I think we should mark all traits or none to keep a consistency.

File: akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala (hunk 2 above)
Comment (nikom1337) [reply akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala]: actualValue is not included in error message anymore + I used new rejectRequestEntityAndComplete method to handle InvalidRequiredValueForQueryParamRejection

File: akka-http/src/main/scala/akka/http/javadsl/server/Rejections.scala (hunk 1 above)
Comment (jlprat) [reply akka-http/src/main/scala/akka/http/javadsl/server/Rejections.scala]: In my opinion, I would put `@DoNotInherit` to all of them

**Comment by jlprat** [comment]
Failure was #1766 

PLS BUILD

**Comment by jlprat** [comment]
@raboof are the changes satisfying your comments? Can we merge?

**Pull request merged**